### PR TITLE
MAT-6935 extracting the version from the url

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-6935](https://jira.cms.gov/browse/MAT-6935)
(Optional) Related Tickets:

### Summary
extracting the version from the url

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java [hunk 1]
```diff
@@ -163,13 +165,25 @@ public void enterCodesystemDefinition(cqlParser.CodesystemDefinitionContext ctx)
...
+      codeSystem.setCodeSystemVersion(getParsedVersion(csDef.getVersion()));
       codeSystem.setCodeSystemName(csDef.getName());
 
       codeSystemMap.putIfAbsent(identifier, codeSystem);
     }
   }
 
+  private String getParsedVersion(String version){
+    if(version != null) {
+      String versionRegex= "urn:hl7:version:(\\d+(\\.\\d+)?)";
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java]: Will this capture versions containing hyphens or multiple periods, e.g. SNOMED's 2024-03 and ActCode's 9.0.0?

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java:1]: Looks like the formatter didn't run. Check if you have any changes post-push that came from the formatter.

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java (hunk 1 above)
Comment (jkotanchik-SB) [reply src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java]: An alternative approach is to use substring since the version prefix is a constant and the value always appears in the same location:
```csDef.getVersion().substring("urn:hl7:version:".length())```

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java
Comment (sb-prateekkeerthi) [reply src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java:1]: I have just run the formatter and push the code along with other changes

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java]: Thank you, I just pushed the code with the changes

**Review by jkotanchik-SB — APPROVED** [review]
